### ISSUE TITLE
Qualificação do Mestrado

### ISSUE THREAD

**Issue by vitalwarley** [vitalwarley/research#84]
Deadline: 16/08

**Comment by vitalwarley** [comment]
Primeira versão da introdução foi finalizada ontem, faltando apenas o outline do documento (i.e. dissertation structure, mas acho que vai ser dispensável). Hoje/agora acabei a primeira versão dos trabalhos relacionados. 

![image](https://github.com/user-attachments/assets/284824ca-02fb-431a-b4de-22b9d02dfebb)

As seções que seguem (Methodology, Experiments and Results, Discussion, Conclusion (and Future Work), e Cronogram) estão no planejamento passado. Planejei começar a escrita semana passada, mas comecei ontem. A ideia era finalizar a escrita dia 16/08. Se preciso, reajusto o planejamento.

**Comment by vitalwarley** [comment]
@tfvieira, o que acha da seção de fundamentação teórica abaixo?

1. Deep Learning in Computer Vision
   - Convolutional Neural Networks (CNNs)
   - Transfer learning and pre-trained models
   - Attention mechanisms
   - Transformer architectures

2. Face Recognition Techniques
   - Overview of face recognition methods
   - Face recognition architectures (including AdaFace and others)

3. Feature Extraction and Representation
   - Importance in kinship verification
   - Deep feature extraction methods
   - Architectures for similarity learning (e.g., Siamese networks, triplet networks)

4. Contrastive Learning
   - Principles of contrastive learning
   - Contrastive loss: original formulation
   - Applications in similarity-based tasks
   - Recent developments in contrastive learning

5. Data Augmentation and Hard Sample Generation
   - Concept and importance in training
   - Various augmentation techniques
   - Feature-space augmentation
   - Hard sample generation methods

6. Current State-of-the-Art in Kinship Verification
   - Recent deep learning approaches [já fiz a primeira versão]
   - Benchmarks and datasets

**Comment by vitalwarley** [comment]
@tfvieira, o que acha da seção de fundamentação teórica abaixo?

1. Deep Learning in Computer Vision
   - Overview
   - Face Recognition
   - Feature Extraction and Representation
   
2. Contrastive Learning
   - Principles of contrastive learning
   - Contrastive loss: original formulation
   - Applications in similarity-based tasks
   - Recent developments in contrastive learning

3. Data Augmentation and Hard Sample Generation
   - Concept and importance in training
   - Various augmentation techniques
   - Feature-space augmentation
   - Hard sample generation methods

5. Current State-of-the-Art in Kinship Verification
   - Recent deep learning approaches [já fiz a primeira versão]
   - Benchmarks and datasets

**Comment by vitalwarley** [comment]
> 1. Deep Learning in Computer Vision
>    * Overview
>    * Feature Extraction and Representation
>    * Face Recognition
>    * Kinship Recognition

Versão inicial finalizada, mas não revisada manualmente.

**Comment by vitalwarley** [comment]
Revisei as seções _Overview_ e _Feature Extraction and Representation_. Adicionei citações, complementei com imagens e mais detalhes.

**Comment by vitalwarley** [comment]
Modifiquei _Face Recognition_, adicionando uma figura e explicando sucintamente os modelos mais importantes e recentes (ArcFace, MagFace e AdaFace).

**Comment by vitalwarley** [comment]
Modifiquei _Kinship Recognition_, adicionando duas figuras, citações e uma tabela com datasets.

**Comment by vitalwarley** [comment]
Versão inicial de 2.2 Contrastive Learning > 2.2.1 Overview foi adicionada. Talvez esteja técnica demais. Em breve reviso.

**Comment by vitalwarley** [comment]
Versão inicial de _2.2 Contrastive Learning > 2.2.2 Key components_ foi adicionada. 

Essa seção 2.2. ainda vai ser revisada. Não estou satisfeito com o fluxo de ideias.

**Comment by vitalwarley** [comment]
Atualmente temos 

- 5 páginas de Introdução
- 22 páginas de Fundamentação Teórica (inclui trabalhos relacionados)

**Comment by vitalwarley** [comment]
> Versão inicial de _2.2 Contrastive Learning > 2.2.2 Key components_ foi adicionada.
> 
> Essa seção 2.2. ainda vai ser revisada. Não estou satisfeito com o fluxo de ideias e certamente faltam algumas coisas.

Reescrevi quase completamente.

![image](https://github.com/user-attachments/assets/07305f25-a60c-4a4c-a8cb-77819cb66a85)


> Atualmente temos
> 
> * 5 páginas de Introdução
> * **24** páginas de Fundamentação Teórica (inclui trabalhos relacionados)

**Comment by vitalwarley** [comment]
Versões iniciais de _Methods_ e _Cronogram_ OK.

**Comment by vitalwarley** [comment]
Trabalhos feitos para PPGI021 foram adicionados

- Resultados preliminares até 06/2023
- Revisão Sistemática

**Comment by vitalwarley** [comment]
_Experiments and Preliminary Results_ OK. 1-36 páginas totais de conteúdo. Total de 76 páginas de todo o documento, incluindo anexos.

O que falta? Editar, resolver erros, adicionar acrônimos e afins.

**Comment by vitalwarley** [comment]
- Adicionei título
- Adicionei nova data
- Diminui legenda da figura 6 -- movi a legenda (refraseada) pro texto e coloquei um sumário (da original) no lugar.
- Melhorei a legenda da figura 12
- Adicionei abreviações e movi página do fim para o começo do documento, substituindo a antiga página customizada "Lista de Abreviaturas e Siglas"
  - Por agora apenas substitui os acrônimos da **Introdução**.
  
@tfvieira, 

- A capa precisa também ser em inglês?
- Avalia meus TODOs e confere se são realmente necessários, por favor.

**Comment by vitalwarley** [comment]
- Adicionei o restante dos acrônimos, até onde consegui notar.
- Alterei pontualmente alguns trechos, dado que não era o foco.

Próximo passo (ainda hoje) é revisar a corretude e flow do texto. Em seguida, preparar a apresentação para sexta no grupo zoioz.

**Comment by vitalwarley** [comment]
Revisão do texto finalizada. Certamente ainda há melhorias a serem feitas, como melhorar a seção _Sample Optimization_ em clareza e didática. Todavia, fica para outro momento. NEssa revisão, além de adicionar acrônimos faltantes, reescrevi e adicionei parágrafos, bem como adicionei figuras ilustrativas (e.g. duas figuras foram adicionadas na seção _Training Dynamics_).

Agora tratar da apresentação para sexta.

**Comment by vitalwarley** [comment]
Versão inicial da apresentação finalizada, mas ainda há o que corrigir, (e.g. figuras), adicionar (e.g. conteúdos importantes), remover (e.g. conteúdos supérfulos).

**Comment by vitalwarley** [comment]
- Introduction OK -- algumas adições na ordem dos conceitos, alguns outros ajustes.
- Theoretical Foundations
  - Removi exemplo de convolução e pooling dos slides.  
  - Removi "Kinship Recognition Tasks" porque o foco é RFIW Tasks msm.
  - Adicionei slide com fórmula para InfoNCE
  - Removi slides sobre contexto histórico, perdas contrastivas iniciais, e perdas contrastivas recentes.
  - Adicionei slide com fórmula para SupCon
  - Dividi Training Dynamic em dois slides, adicionado mais uma figura
  - Condensei Negative Sampling e Feature Transformation no slide Sample Optimization
  - Mantive apenas 3 trabalhos relacionados principais
- Methods
  - Removi slide "Methods Overview", que apenas reafirma objetivos
  - Removi slide "Development Process - Key Stages", que descreve os estágios da pesquisa; descreverei oralmente no próximo slide que o segue.
- Experiments and Preliminary Results
  - Removi slide de overview -- já vou apresentando msm
  - Juntei slides com resultados de experimentos e figura
  - Juntei slides com resultados finais e tabela

Essa edição deixou a apresentação com total de 46 páginas, mas apenas 34 são de conteúdo. Farei mais uma revisão em breve; ainda falta definir as fórmulas de HCL e PFE também.

**Comment by vitalwarley** [comment]
Removi "Contrastive Learning Example". Agreguei outros slides em um só "Contrastive Learning". Removi "Training and Inference Pipelines" de Face Recognition. TIrei também "Dissertation Structure", já que o ToC é mostrado. Tirei o slide sobre os datasets, já que não traz informação à apresentação. Juntei os dois sobre "Experimental Setup" em um só.

Agora temos 36 slides, mas uns 24 de conteúdo.

**Comment by vitalwarley** [comment]
Acabei expandindo "Sample Optimization" em três, dado a importância desse conceito. Final deu 38 slides, com 28 de conteúdo.

**Comment by vitalwarley** [comment]
Condensei as sugestões do grupo nas tarefas abaixo, aliado a algumas de meu interesse, e pretendo terminar em duas semanas (em 30/08 ter uma versão 2 da qualificação e slides).

![image](https://github.com/user-attachments/assets/307d4d55-7c81-4c0b-833f-7ef8959c7acf)

**Comment by vitalwarley** [comment]
Atualize boa parte da introdução, onde adicionei figuras representativas aos desafios, bem como movi a seção de trabalhos relacionados da fundamentação teórica para cá. Alterei algumas frases, removi outras etc. Ainda tenho dúvidas, no entanto, quanto às subseções: objetivos, proposta e relevância. Estão OK? Objetivos estão gerais demais? Proposta está específica demais? Relevância foca devidamente na proposta sob o ponto de vista geral da área?

**Comment by vitalwarley** [comment]
Refiz as seções de objetivos, proposta e relevância. Deixei os objetivos mais claros sobre o que pretendo, enquanto tornei a proposta mais abstrata (não citei os métodos que usei, mas os abstrai em termo do que resolvem; os métodos cito melhor nos resultados preliminares), e ajustei a relevância pra enfatizar o que podemos conseguir com a satisfação dos objetivos (foco no impacto mais que na metodologia).

**Comment by vitalwarley** [comment]
Atualizei a seção de Métodos para se adequar aos ajustes da Introdução.

**Comment by vitalwarley** [comment]
Estou atualizando a seção de experimentos e resultados preliminares. O número de amostras que tenho no total é 26k, mas Survey on the Analysis and Modeling of Visual Kinship: A Decade in the Making diz que há 33k. Os dados que tenho são da [FaCoRNet](https://github.com/wtnthu/FaCoR), creio que os mesmos do [SOTA RFIW2021](https://github.com/Zhangxm520/RFIW2021). Não consegui verificar porque o link para o drive que colocam no repositório está quebrado. Os dados disponibilizados pelo SOTA no RFIW2021, Zhang et al. (2021), são realmente 26k -- pelo menos os que tenho aqui baixados.

Talvez usaram uma versão antiga? Zhang et al. (2021) foi publicado no FG2021, cujo 1o round foi em 05/2021 e o 2o em 08/2021. O FIW foi atualizado já nesse RFIW2021, porque o SOTA no RFIW2020 foi o Shadrikov (2020) e ele cita

> The Recognizing Families In the Wild Data Challenge (RFIW2020) focuses on determining blood relations based on visual facial similarities. For that, it has 20867 images from 763 families for training and validation sets.

**Comment by vitalwarley** [comment]
Atualizei a seção de experimentos e resultados preliminares e adicionei uma seção de conclusão. Próximo passo é refazer os experimentos -- baseline e os demais. Pretendia reproduzir FaCoRNet novamente, mas não vou fazer porque os resultados foram diferentes dos de Abril e não há tempo para entender o porquê. Usarei os resultados reportados mesmo.

**Comment by vitalwarley** [comment]
Desde terça venho refazendo experimentos em prol da ablação e reprodutibilidade. Notei que o componente Positive Feature Extrapolation não foi corretamente implementado por mim e, quando reportei os resultados, acredito que comparei entre os conjuntos errados -- validação v1 em vez do teste set. Desde então venho fazendo correções e adições e acredito que cheguei a uma versão final, que também tem a interpolação dos negativos. Até o momento, entendi que infelizmente a transformação de atributos aparenta não trazer ganhos. No máximo, a interpolação de positivos ($\lambda_ex < 1) aparenta trazer mais variedade nas amostras e, consequentemente, uma convergência mais lenta, embora não necessariamente melhores resultados. Pretendo fechar o artigo ainda hoje, mas apenas com a HCL.

**Comment by vitalwarley** [comment]
Outro problema que não entendo a razão é o t-SNE / UMAP

**RFIW2021 paper** 
![image](https://github.com/user-attachments/assets/b9212afa-3225-4d9e-bcfe-aaf0110ae97d)

**RFIW2021 weights**

Treinei ano passado, se n me engano, usando o código dos autores. Abaixo t-SNE com as embeddings de **projeção**

![image](https://github.com/user-attachments/assets/8973fa89-5b7d-4dd4-917b-f192b66ee28c)

Abaixo t-SNE com as embeddings do **backbone**

![image](https://github.com/user-attachments/assets/d42e5b0b-2b01-4e28-9821-61301cde3e8f)

**AdaFace + Sampler + HCL**

![image](https://github.com/user-attachments/assets/2909743d-c3cc-458b-b4d7-3362c215a963)

**Comment by vitalwarley** [comment]
Até ajustei o código para treinar novos modelos usando o código do RFIW2021 como base, mas consegui.

**Comment by vitalwarley** [comment]
Escalei as imagens para [0, 1] e...

*cries deeply*

![image](https://github.com/user-attachments/assets/9f297471-008e-49d0-b4d7-bdcc5a804444)

**Comment by vitalwarley** [comment]
Postando aqui esses gráficos porque estou orgulhoso do que fiz. Essa visualização permite ver claramente os impacto dos diferentes componentes usados (e.g. dataset, temperatura da perda, presença da perda HD).

![image](https://github.com/user-attachments/assets/0bbfee09-cd99-408d-8439-fddec15d5fd4)

**Comment by vitalwarley** [comment]
Agora o que resta é reescrever a seção de experimentos e resultados. O que fiz, o que consegui, o que significa.

**Comment by vitalwarley** [comment]
I think I am done. **59 páginas de conteúdo**.  101 páginas totais. Se amanhã houver tempo, talvez possa melhora a seção 2 Theoretical Foundations, especificamente 2.2.5 Training Dynamics e 2.2.6 Sample Optimization.

**Comment by vitalwarley** [comment]
https://github.com/vitalwarley/research/issues/38#issuecomment-2495454002
